### PR TITLE
Kinda fix swap screen keyboard jittering

### PR DESCRIPTION
**Pull request by terrysahaidak**
Fixes TEAM1-106
Figma link (if any):

## What changed (plus any additional context for devs)

The problem with jittering was related to a different keyboard height. Since we render the screen with one keyboard, navigate away and render the keyboard again with a different height - it jumps to the new height. There is no way to actually know what the keyboard height is gonna be until it's open. Also, everything is async, so even if we knew - it would be too late.

Now we're handling the keyboard on Android in a different way. The Keyboard Area module is our best friend. 
But also, we remember what was the tallest keyboard ever during the session and try to push the floating modal up so it won't be covered, but also we account for the screen height and the content height, so we won't push it outside the screen. 
If the screen is too small - we have to cover the view with the keyboard to fit everything.

Also, this PR makes the Swap screen look the same as on iOS.

## Screen recordings / screenshots

https://user-images.githubusercontent.com/7809008/185684495-811f9057-08ff-4c67-a0ea-f6d719a8c394.mp4

## What to test

- Swap screen
- Select currency screen
- It should not jump, the view should not be "outside" the screen

## Final checklist

- [x] Assigned individual reviewers?
- [x] Added labels? (`team1/team2`, `critical path`, `release`, `dev QA`)
- [x] Did you test both iOS and Android?
- [x] If your changes are visual, did you check both the light and dark themes?
- [ ] Added e2e tests? If not, please specify why
- [ ] If you added new critical path files, did you update the CODEOWNERS file?
- [ ] If no `dev QA` label, did you add the PR to the QA Queue?

### CODE REVIEW

**Review by perunt — APPROVED** [review]
It looks good to me, at least as far as Android could be now with a flaky keyboard

File: src/screens/CurrencySelectModal.js [hunk 1]
```diff
@@ -64,8 +71,19 @@ const TabTransitionAnimation = styled(Animated.View)(
   position.sizeAsObject('100%')
 );
 
+function AndroidWrapper({ children }) {
+  const insets = useSafeArea();
+  const keyboardHeight = useKeyboardArea();
+  const marginBottom = keyboardHeight > 0 ? insets.top + 4 + keyboardHeight : 0;
```
Comment (tchayen) [inline src/screens/CurrencySelectModal.js:77]: Why 4?

**Review by tchayen — APPROVED** [review]
Works fine. Changes are scary, but I don't see any other way of making it work.

I left some small comments but nothing blocking.

File: src/screens/CurrencySelectModal.js (hunk 1 above)
Comment (brunobar79) [reply src/screens/CurrencySelectModal.js:77]: I'd just move the 4 to a const with the name that represent it better (like androidPadding) or whatever

**Review by brunobar79 — APPROVED** [review]
🚀 🚀 🚀

File: src/screens/CurrencySelectModal.js (hunk 1 above)
Comment (terrysahaidak) [reply src/screens/CurrencySelectModal.js:77]: 4 is basically little padding that iOS has, without it, the bottom part will be at the same height as the keyboard. will move it to the constant.

File: ios/Podfile.lock [hunk 2]
```diff
@@ -1019,7 +1019,7 @@ SPEC CHECKSUMS:
   FirebaseRemoteConfig: a5d9188d8f57f602636f111eca6460ebe32b9b71
   FLAnimatedImage: 8f3e854ebf9680b71f0b6e700939687fc4d82b4b
   fmt: ff9d55029c625d3757ed641535fd4a75fedc7ce9
-  glog: 476ee3e89abb49e07f822b48323c51c57124b572
+  glog: 5337263514dd6f09803962437687240c5dc39aa4
```
[the remark this answers is not in the document]
Comment (terrysahaidak) [reply ios/Podfile.lock:1022]: yeah could be

File: src/screens/CurrencySelectModal.js [hunk 3]
```diff
@@ -60,12 +67,38 @@ const getHasShownWarning = () =>
...
+
+  const marginBottom = keyboardHeight > 0 
+    // tiny padding adds a tiny space between the keyboard and the view
+    //  to match how it looks on iOS
+    ? insets.top + TINY_ANDROID_PADDING + keyboardHeight 
+    : 0;
+  return (
+    <Flex>
+      <Flex style={{ marginBottom }}>{children}</Flex>
+    </View>
```
Comment (estebanmino) [inline src/screens/CurrencySelectModal.js:96]: this is crashing the app, you meant to use `View` as the opening tag as well? is `Flex` right now

Comment (estebanmino) [reply src/screens/CurrencySelectModal.js:96]: i replaced View as open tag and i'm seeing this https://www.loom.com/share/f488df4bd57d4b21988ccd24ded6b20e not sure if that's what you had here before

Comment (terrysahaidak) [reply src/screens/CurrencySelectModal.js:96]: Yes, typo, gonna fix that

**Review by estebanmino — COMMENTED** [review]
looking better now! https://www.loom.com/share/05dc91eb8a124bb28efab82a57d7bd9d

on android i see the fee section has a margin with the keyboard when no output token selected, it's fine when the output token is selected since we have settings and flip buttons there
image

**Pull request closed without merging**